### ISSUE TITLE
Ministart start fails when running from drive other than user profile in Windows

### ISSUE THREAD

**Issue by budhrg** [minishift/minishift#236]
`minishift start` fails when running from drive other than `%USERPROFILE%` ( C drive mostly ).

```powershell
PS D:\> .\minishift.exe version
minishift version: 1.0.0-beta.1

PS D:\> .\minishift.exe start --vm-driver virtualbox -v 5 --show-libmachine-logs
Starting local OpenShift cluster...
Found binary path at D:\minishift.exe
Launching plugin server for driver virtualbox
Plugin server listening at address 127.0.0.1:49698
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(minishift) Calling .GetMachineName
(minishift) Calling .DriverName
Running pre-create checks...
(minishift) Calling .PreCreateCheck
(minishift) DBG | COMMAND: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe --version
(minishift) DBG | STDOUT:
(minishift) DBG | {
(minishift) DBG | 5.0.30r112061
(minishift) DBG | }
(minishift) DBG | STDERR:
(minishift) DBG | {
(minishift) DBG | }
(minishift) DBG | Hyper-V is not installed.
(minishift) DBG | %!(EXTRA *exec.Error=exec: "vmms.exe": executable file not found in %PATH%)COMMAND: wmic cpu get VirtualizationFirmwareEnabled
(minishift) DBG | STDOUT:
(minishift) DBG | {
(minishift) DBG | VirtualizationFirmwareEnabled
(minishift) DBG | TRUE
(minishift) DBG |
(minishift) DBG | }
(minishift) DBG | COMMAND: C:\Program Files\Oracle\VirtualBox\VBoxManage.exe list hostonlyifs
(minishift) DBG | STDOUT:
(minishift) DBG | {
(minishift) DBG | Name:            VirtualBox Host-Only Ethernet Adapter #2
(minishift) DBG | GUID:            86707bde-8f13-4f5d-8381-f953689021cf
(minishift) DBG | DHCP:            Disabled
(minishift) DBG | IPAddress:       192.168.99.1
(minishift) DBG | NetworkMask:     255.255.255.0
(minishift) Calling .GetConfigRaw
(minishift) DBG | IPV6Address:
(minishift) DBG | IPV6NetworkMaskPrefixLength: 0
(minishift) DBG | HardwareAddress: 0a:00:27:00:00:0c
(minishift) DBG | MediumType:      Ethernet
(minishift) DBG | Status:          Up
(minishift) DBG | VBoxNetworkName: HostInterfaceNetworking-VirtualBox Host-Only Ethernet Adapter #2
(minishift) DBG |
(minishift) DBG | }
Creating machine...
(minishift) DBG | STDERR:
(minishift) DBG | {
(minishift) Calling .Create
(minishift) DBG | }
(minishift) Downloading C:\Users\budhram\.minishift\cache\boot2docker.iso from file://C:/Users/budhram/.minishift/cache/iso/boot2docker.iso...
E1230 13:54:26.113439    5848 start.go:130] Error starting host: Error creating. Error creating machine: Error in driver during machine creation: open /Users/budhram/.minishift/cache/iso/boot2
```

Following similar issues are reported in `minikube`:
- https://github.com/kubernetes/minikube/issues/459
- https://github.com/kubernetes/minikube/issues/622 (dup of above #459)
- https://github.com/kubernetes/minikube/issues/848 (dup of above #459)

**Comment by budhrg** [comment]
Reason explained here https://github.com/kubernetes/minikube/issues/459#issuecomment-240796185.
Something feel like a limitation to me at the moment.

**Comment by apitt** [comment]
Is there any progress towards fixing this?  This is a blocker where I work.  We don't have control of where %userprofile% is mapped, and we won't be able to run minishift until this issue is resolved.

Windows 7 with VirtualBox 5.1.18.

**Comment by praveenkumar** [comment]
@budhrg Can we close this now?

**Comment by hferentschik** [comment]
@praveenkumar I think this is still an issue. There is afaiu no solution for this for now.

**Comment by budhrg** [comment]
> I think this is still an issue. There is afaiu no solution for this for now.

Might be we need to do bit work in finding an alternate solution here. Will discuss with @gbraad

**Comment by gbraad** [comment]
> So the solution was ... to be on the drive the Users directory is located on
https://github.com/kubernetes/minikube/issues/459#issuecomment-240796185

**Comment by apitt** [comment]
If the documented solution for people like me (a corporate desktop setup that defaults the Users directory to some network drive) is to set the `HOMEDRIVE `environment variable to `C:`, then I think it can be closed.  If it's not documented, then people will probably continue to open this issue.

Also, this happens to work for us (Windows 7 and Virtualbox), I can't really vouch for this being a "universal" solution.

**Comment by hferentschik** [comment]
Well, tbh I would like to understand why it is not possible to do it. This seems like an very odd restriction imo

**Comment by gbraad** [comment]
Creating a new issue as the location has to be `C:\` regardless of the `%USERPROFILE% as is currently documented.

**Comment by hferentschik** [comment]
> Creating a new issue as the location has to be C:\ regardless of the `%USERPROFILE% as is currently documented.

+1

given that we get multiple error reports around this, I think we should investigate the cause for this asap.

**Comment by praveenkumar** [comment]
@minishift/minishift-dev should we close this issue since we already have documents in place?

**Comment by budhrg** [comment]
Closing this issue as it can't be resolved from Minishift side.

It need changes from docker/machine side at

https://github.com/docker/machine/blob/1e3b6b64504f2970252b06119d6dc1fb1982b4b4/libmachine/mcnutils/b2d.go#L195-L204

**Comment by gbraad** [comment]
> It need changes from docker/machine side at
https://github.com/docker/machine/blob/1e3b6b64504f2970252b06119d6dc1fb1982b4b4/libmachine/mcnutils/b2d.go#L195-L204

if changes are needed, we can now make them at machine-drivers/machine

**Comment by la-idea** [comment]
I solved it in my case. Well, I made a mistake when I was setting up. Instead of adding PATH to the environment variable, I replaced all Windows PATH adresses. Without this PATH, cmd does not execute commands. This is the default PATH C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem; More than one path separated by ';'. You just have to add the path minishift or mindocker after the ';'. It worked for me.
